encode.go

```diff
@@ -24,6 +24,35 @@ import (
 	perrors "github.com/pkg/errors"
 )
 
+var listTypeName = map[string]string{
```

Comment:
adding the definition in file list.go will be better

Reply:
or it should be a part of pojo.  it does not need to maintain a list name mapping, instead a type name mapping is needed, when encodeing, just need add a prefix `[`.

Reply:
This map only works in the list. I thinks it should be in the list.go .

Reply:
Should we add a rwlock on `listTypeName`? Is there any concurrency problem with it?

Reply:
“registerTypeName” be called when calling “RegisterPOJO”， so, I think there is not any consurrency problem.

Reply:
@fangyincheng There are multidimensional arrays, like `[[[string`， which are also need to be registered automatically, see an example that [extracting type from object](https://github.com/vogo/gohessian/blob/b55c95c7e1f6aa4721739e9f4359c62c2034e536/reflection.go#L53)

Reply:
@wongoo I think I need add support for multidimensional arrays.

Reply:
@fangyincheng  ok, waiting for your new progress